Mail/Views/Bottom sheets/DiscoveryView.swift

```diff
@@ -0,0 +1,170 @@
+/*
+ Infomaniak Mail - iOS App
+ Copyright (C) 2022 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import InfomaniakCoreUI
+import InfomaniakDI
+import MailCore
+import MailResources
+import SwiftUI
+
+public struct DiscoveryItem {
+    public let image: MailResourcesImages
+    public let title: String
+    public let description: String
+    public let primaryButtonLabel: String
+    public let matomoCategory: MatomoUtils.EventCategory
+}
+
+public extension DiscoveryItem {
+    static let aiDiscovery = DiscoveryItem(
+        image: MailResourcesAsset.aiIllustration,
+        title: MailResourcesStrings.Localizable.aiDiscoveryTitle,
+        description: MailResourcesStrings.Localizable.aiDiscoveryDescription,
+        primaryButtonLabel: MailResourcesStrings.Localizable.buttonTry,
+        matomoCategory: .aiWriter
+    )
+
+    static let syncDiscovery = DiscoveryItem(
+        image: MailResourcesAsset.syncIllustration,
+        title: MailResourcesStrings.Localizable.syncCalendarsAndContactsTitle,
+        description: MailResourcesStrings.Localizable.syncCalendarsAndContactsDescriptioniOS,
+        primaryButtonLabel: MailResourcesStrings.Localizable.buttonStart,
+        matomoCategory: .syncAutoConfig
+    )
+
+    static let updateDiscovery = DiscoveryItem(
+        image: MailResourcesAsset.logoMailWithStar,
+        title: MailResourcesStrings.Localizable.updateAvailableTitle,
+        description: MailResourcesStrings.Localizable.updateAvailableDescription,
+        primaryButtonLabel: MailResourcesStrings.Localizable.buttonUpdate,
+        matomoCategory: .appUpdate
+    )
+}
+
+struct DiscoveryView: View {
+    @LazyInjectService private var matomo: MatomoUtils
+
+    @Environment(\.isCompactWindow) private var isCompactWindow
+    @Environment(\.dismiss) private var dismiss
+
+    let item: DiscoveryItem
+
+    @State private var willDiscoverNewFeature = false
+
+    let shouldPresentFeature: (Bool) -> Void
```

Comment:
I don't understand the name of this parameter.
As it's called on the `onAppear()` of the view, we dont know if we should present the feature?
Maybe we could call it `onAppear`?